mermithid/misc/FakeTritiumDataFunctions.py

```diff
@@ -0,0 +1,655 @@
+    interp_error = interp1d(efficiency_dict['frequencies'], np.mean(efficiency_dict['error interp with slope correction'], axis=0), fill_value=1, bounds_error=False)
+    return interp_efficiency(f), interp_error(f)
+
+
+##############################################
+"""
+functions from detailed lineshape
+"""
+
+"""
+Adapted from https://github.com/project8/scripts/blob/feature/KrScatterFit/machado/fitting/data_fitting_rebuild.py#L429 - by E. M. Machado
+
+Changes from that version:
+- Kr specific functions (shake-up/shake-off) removed
+- Spectrum is normalized
+"""
+
+
+# Natural constants
+kr_line = 17.8260 # keV
+kr_line_width = 2.83 # eV
+e_charge = 1.60217662*10**(-19) # Coulombs , charge of electron
+m_e = 9.10938356*10**(-31) # Kilograms , mass of electron
+mass_energy_electron = 510.9989461 # keV
+max_scatters = 20
+gases = ["H2"]
+
+# This is an important parameter which determines how finely resolved
+# the scatter calculations are. 10000 seems to produce a stable fit, with minimal slowdown
+num_points_in_std_array = 10000
+
+# Establishes a standard energy loss array (SELA) from -1000 eV to 1000 eV
+# with number of points equal to num_points_in_std_array. All convolutions
+# will be carried out on this particular discretization
+def std_eV_array():
+    emin = -1000
+    emax = 1000
+    array = np.linspace(emin,emax,num_points_in_std_array)
+    return array
+
+# A lorentzian function
+def lorentzian(x_array,x0,FWHM):
+    HWHM = FWHM/2.
+    func = (1./np.pi)*HWHM/((x_array-x0)**2.+HWHM**2.)
+    return func
+
+# A lorentzian line centered at 0 eV, with 2.83 eV width on the SELA
+def std_lorentzian_17keV():
+    x_array = std_eV_array()
+    ans = lorentzian(x_array,0,kr_line_width)
+    return ans
+
+
+# A gaussian function
+def gaussian(x_array,A,sigma=0,mu=0):
+    if isinstance(A, list):
+        a = A
+        x = x_array
+        return 1/((2.*np.pi)**0.5*a[0])*(np.exp(-0.5*((x-a[1])/a[0])**2))
+    else:
+        f = A*(1./(sigma*np.sqrt(2*np.pi)))*np.exp(-(((x_array-mu)/sigma)**2.)/2.)
+        return f
+
+# A gaussian centered at 0 eV with variable width, on the SELA
+def std_gaussian(sigma):
+    x_array = std_eV_array()
+    ans = gaussian(x_array,1,sigma,0)
+    return ans
+
+# Converts a gaussian's FWHM to sigma
+def gaussian_FWHM_to_sigma(FWHM):
+    sigma = FWHM/(2*np.sqrt(2*np.log(2)))
+    return sigma
+
+# Converts a gaussian's sigma to FWHM
+def gaussian_sigma_to_FWHM(sigma):
+    FWHM = sigma*(2*np.sqrt(2*np.log(2)))
+    return FWHM
+
+# normalizes a function, but depends on binning.
+# Only to be used for functions evaluated on the SELA
+def normalize(f):
+    x_arr = std_eV_array()
+    f_norm = integrate.simps(f,x=x_arr)
+    # if  f_norm < 0.99 or f_norm > 1.01:
+    #     print(f_norm)
+    f_normed = f/f_norm
+    return f_normed
+
+#returns array with energy loss/ oscillator strength data
+def read_oscillator_str_file(filename):
```

Comment:
This function should be used during the configuration of the processor to make sure the files are present and readable. If possible all the energyOsc variables (and everything useful later in the calculation) should be saved inside the processor (as self.energyOsc) to be available for the Run part of the processor (without having to read the file again).

Reply:
This function is not actually used. The data generation requires that the convolutions were pre-calculated and saved. I commented all unused functions for now.

Reply:
I also moved the lineshape functions to KrLineshapeFunctions.py